packages/mcl/src/src/mcl/commands/compare_disko.d

```diff
@@ -0,0 +1,188 @@
+            return "🟩";
+            break;
+        case ChangeStatus.Changed:
+            return "⚡";
+            break;
+        case ChangeStatus.Removed:
+            return "🗑";
+            break;
+        case ChangeStatus.New:
+            return "🧩";
+            break;
+    }
+}
+
+// TODO: handle case where a machine is missing from one branch
+export void compare_disko(string[] args){
+    const params = parseEnv!Params;
+
+    JSONValue configurations = nix.flake!JSONValue("", [], "show");
+
+    string[] machinesNew;
+    foreach (string k, JSONValue v; configurations["nixosConfigurations"]){
+        if (k[$-3 .. $] != "-vm"            &&
+            k != "gitlab-runner-container"  &&
+            k != "minimal-container"        )
+        {
+            machinesNew ~= k;
+        }
+    }
+
+    auto attr = constructCommandAttr("./.", machinesNew);
+    auto machineOptionsRootNew = nix.eval!JSONValue("", ["--impure", "--expr", attr]);
+
+
+    string gitRoot = getTopLevel();
+    string worktreeBaseBranch = gitRoot~"-"~params.baseBranch;
+
+    if (execute("git rev-parse --abbrev-ref HEAD") == params.baseBranch){
+        errorAndExit("Trying to compare branch "~params.baseBranch~" with itself. Quitting.");
+    }
+
+    execute(["git", "worktree" , "add", worktreeBaseBranch, params.baseBranch]);
+
+    configurations = nix.flake!JSONValue(worktreeBaseBranch, [], "show");
+
+    string[] machinesOld;
+    foreach (string k, JSONValue v; configurations["nixosConfigurations"]){
+        if (k[$-3 .. $] != "-vm"            &&
+            k != "gitlab-runner-container"  &&
+            k != "minimal-container"        )
+        {
+            machinesOld ~= k;
+        }
+    }
+
+    attr = constructCommandAttr(worktreeBaseBranch, machinesOld);
+    auto machineOptionsRootOld = nix.eval!JSONValue("", ["--impure", "--expr", attr]);
+
+    MachineChanges[] machineChanges;
+
+    string[] machines = uniq(machinesOld ~ machinesNew).array;
+
+    foreach (string m; machines){
+        MachineChanges mc;
+        mc.machine = m;
+        if (m in machineOptionsRootOld && !(m in machineOptionsRootNew)){
+            static foreach (setting; diskoOptions){
+                __traits(getMember, mc, setting) = ChangeStatus.Removed;
+            }
+        }
+        else if (m in machineOptionsRootNew && !(m in machineOptionsRootOld)){
+            static foreach (setting; diskoOptions){
+                __traits(getMember, mc, setting) = ChangeStatus.New;
+            }
+        }
+        else{
+            static foreach (setting; diskoOptions){
+                if (machineOptionsRootOld[m][setting] == machineOptionsRootNew[m][setting]){
+                    __traits(getMember, mc, setting) = ChangeStatus.Unchanged;
+                }
+                else{
+                    __traits(getMember, mc, setting) = ChangeStatus.Changed;
+                }
+            }
+        }
+        machineChanges ~= mc;
+    }
```

Comment:
Consider using appender!string instead of "~", as it's faster and more efficient.
https://forum.dlang.org/thread/ppbczcbqhaqalgwcjbba@forum.dlang.org